optuna/samplers/_tpe/probability_distributions.py

```diff
@@ -108,38 +108,43 @@ def sample(self, rng: np.random.RandomState, batch_size: int) -> np.ndarray:
 
     def log_pdf(self, x: np.ndarray) -> np.ndarray:
         weighted_log_pdf = np.zeros((len(x), len(self.weights)), dtype=np.float64)
+        cont_dists = []
+        cont_inds = []
         for i, d in enumerate(self.distributions):
-            xi = x[:, i]
             if isinstance(d, _BatchedCategoricalDistributions):
-                weighted_log_pdf += np.log(
-                    np.take_along_axis(
-                        d.weights[None, :, :], xi[:, None, None].astype(np.int64), axis=-1
-                    )
-                )[:, :, 0]
+                xi = x[:, i, np.newaxis, np.newaxis].astype(np.int64)
+                weighted_log_pdf += np.log(np.take_along_axis(d.weights[np.newaxis], xi, axis=-1))[
+                    ..., 0
+                ]
             elif isinstance(d, _BatchedTruncNormDistributions):
-                weighted_log_pdf += _truncnorm.logpdf(
-                    x=xi[:, None],
-                    a=(d.low - d.mu[None, :]) / d.sigma[None, :],
-                    b=(d.high - d.mu[None, :]) / d.sigma[None, :],
-                    loc=d.mu[None, :],
-                    scale=d.sigma[None, :],
-                )
+                cont_dists.append(d)
+                cont_inds.append(i)
             elif isinstance(d, _BatchedDiscreteTruncNormDistributions):
-                xi_uniq, xi_inv = np.unique(xi, return_inverse=True)
+                xi_uniq, xi_inv = np.unique(x[:, i], return_inverse=True)
                 mu_uniq, sigma_uniq, mu_sigma_inv = _unique_inverse_2d(d.mu, d.sigma)
-                lower_limit = d.low - d.step / 2
-                upper_limit = d.high + d.step / 2
-                x_lower = np.maximum(xi_uniq - d.step / 2, lower_limit)[:, np.newaxis]
-                x_upper = np.minimum(xi_uniq + d.step / 2, upper_limit)[:, np.newaxis]
```

Comment:
Note: Since `x` is a sampled value, it is within the range of `lower_limit` and `upper_limit`.